specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/examples/assistedQueries/CreateEditedQuery.json

```diff
@@ -0,0 +1,63 @@
+{
+  "parameters": {
+    "api-version": "2022-11-01-preview",
+    "subscriptionId": "d0cfe6b2-9ac0-4464-9919-dccaee2e48c0",
+    "resourceGroupName": "myRg",
+    "workspaceName": "myWorkspace",
+    "operationalInsightsResourceProvider": "Microsoft.OperationalInsights",
+    "queryId": "00000000-0001-4647-9105-6339bfdb4e6a",
+    "editedQueryId": "82ba292c-dc97-4dfc-969d-d4dd9e666842",
+    "editedQuery": {
+      "etag": "0300bf09-0000-0000-0000-5c37296e0000",
+      "properties": {
+        "kustoQueryLanguage": "SystemData | take 20"
```

Comment:
the edited query and query seem to have very similar schemas. Can you please explain why is it modelled as a child of the query ? does an edited query always have to be created in the context of a query ?

Reply:
there will be only one original kql query. the user may edit that original query as much as he/she want and we keep track of the edited queries. the models are simple but we have a parent-child relation since the edited queries are based on one original query